<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CamstreamDialog</class>
 <widget class="QDialog" name="CamstreamDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>320</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QWidget" name="widget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>3</number>
      </property>
      <property name="leftMargin">
       <number>3</number>
      </property>
      <property name="topMargin">
       <number>3</number>
      </property>
      <property name="rightMargin">
       <number>3</number>
      </property>
      <property name="bottomMargin">
       <number>3</number>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Config Name</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="txt_config_name"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>606</width>
        <height>668</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_6">
       <property name="leftMargin">
        <number>3</number>
       </property>
       <property name="topMargin">
        <number>3</number>
       </property>
       <property name="rightMargin">
        <number>3</number>
       </property>
       <property name="bottomMargin">
        <number>3</number>
       </property>
       <property name="spacing">
        <number>3</number>
       </property>
       <item row="19" column="0">
        <widget class="QLabel" name="label_37">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Which network mode to use for the stream. RTSP is generally recommended.&lt;/p&gt;
&lt;ul&gt;
&lt;li&gt;TCP = Pi runs a TCP server that the client connects to. Only allows one client and one stream per port. TCP also gaurentees order of packets, which can result in high latency with slow connection or frequent packet loss.&lt;/li&gt;
&lt;li&gt;UDP = Pi sends frames to a single client using UDP. The address of the client must be enabled ahead of time. UDP does not gaurentee order of packets, so with slow connections the latest frame will always be displayed. The downside is that since no server is run on the Pi, the Pi must know the comptuer's IP address.&lt;/li&gt;
&lt;li&gt;RTSP = Runs an RTSP server on the Pi that clients (one or more) can connect to. Each stream has its own key (path on the server), allowing the same port to be used for multiple streams (each with multiple clients if desired). RTSP will also use UDP transports by default, allowing the benefits of UDP, while still providing a server. A small amount of latency is introduced compared to raw TCP or UDP.&lt;/li&gt;
&lt;/ul&gt;
&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Mode</string>
         </property>
        </widget>
       </item>
       <item row="14" column="2">
        <widget class="QComboBox" name="combox_profile">
         <item>
          <property name="text">
           <string>baseline</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>main</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>high</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="12" column="0">
        <widget class="QLabel" name="label_28">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Video format for the stream. H.264 will likely result in better quailty and higher resolution than MJPEG at the same bandwidth usage, however can increase latency. MJPEG is lower latency, but often much higher bandwidth. The Pi supports hardware H.264 encoding so H.264 is generally recommended.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Format</string>
         </property>
        </widget>
       </item>
       <item row="16" column="2">
        <widget class="QLineEdit" name="txt_bitrate">
         <property name="text">
          <string>2048000</string>
         </property>
        </widget>
       </item>
       <item row="7" column="2">
        <widget class="QLineEdit" name="txt_framerate">
         <property name="toolTip">
          <string>60</string>
         </property>
         <property name="text">
          <string>30</string>
         </property>
        </widget>
       </item>
       <item row="19" column="2">
        <widget class="QComboBox" name="combox_netmode">
         <property name="currentIndex">
          <number>2</number>
         </property>
         <item>
          <property name="text">
           <string>tcp</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>udp</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>rtsp</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_18">
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Source</string>
         </property>
        </widget>
       </item>
       <item row="9" column="2">
        <widget class="QComboBox" name="combox_rotate">
         <item>
          <property name="text">
           <string>0째</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>90째</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>180째</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>270째</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_19">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Which USB device to capture data from (/dev/video#). Only applies for V4L2.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Device</string>
         </property>
        </widget>
       </item>
       <item row="21" column="2">
        <widget class="QLineEdit" name="txt_port">
         <property name="text">
          <string>8554</string>
         </property>
        </widget>
       </item>
       <item row="13" column="2">
        <widget class="QComboBox" name="combox_encoder">
         <item>
          <property name="text">
           <string>libav-omx</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>omx</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>libx264</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="22" column="0">
        <widget class="QLabel" name="label_40">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Path on the RTSP server for this stream. Does not apply in TCP or UDP mode.&lt;/p&gt;&lt;p&gt;Default = stream&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    RTSP Key</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_25">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Height of the video mode. Must be a valid video mode along with the width and framerate settings.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Height</string>
         </property>
        </widget>
       </item>
       <item row="20" column="2">
        <widget class="QLineEdit" name="txt_address">
         <property name="text">
          <string>127.0.0.1</string>
         </property>
        </widget>
       </item>
       <item row="13" column="0">
        <widget class="QLabel" name="label_31">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;When using V4L2 and the H.264 format, several encoders are available.&lt;/p&gt;
&lt;ul&gt;
&lt;li&gt;libav-omx = Hardware accelerated. Works on Buster and newer. (Recommended)&lt;/li&gt;
&lt;li&gt;omx = Hardware accelerated. Doesn't work on Bullseye.&lt;/li&gt;
&lt;li&gt;libx264 = Software only. Will cause higher CPU usage. Not recommended on Pi Zero.&lt;/li&gt;
&lt;ul&gt;
&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Encoder</string>
         </property>
        </widget>
       </item>
       <item row="12" column="2">
        <widget class="QComboBox" name="combox_format">
         <item>
          <property name="text">
           <string>h264</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>mjpeg</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_22">
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Video</string>
         </property>
        </widget>
       </item>
       <item row="17" column="0">
        <widget class="QLabel" name="label_30">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Quality of jpeg compression when using a mjpeg stream. This option has no effect if using an h264 format. This also does not work with the raspicam driver.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Quality</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_26">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Framerate of the video mode. Must be a valid video mode along with the width and height settings.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Framerate</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLineEdit" name="txt_device">
         <property name="text">
          <string>/dev/video0</string>
         </property>
        </widget>
       </item>
       <item row="10" column="0">
        <widget class="QLabel" name="label_35">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Gain of the image. Higher gain makes the image &amp;quot;lighter&amp;quot;. Likely will not work with V4L2.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Gain</string>
         </property>
        </widget>
       </item>
       <item row="11" column="0">
        <widget class="QLabel" name="label_27">
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Stream</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_20">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head /&gt;
&lt;body&gt;
&lt;p&gt;What method is used to read data from the camera.&lt;/p&gt;
&lt;ul&gt;
&lt;li&gt;Raspicam = Pi Camera on Buster and older&lt;/li&gt;
&lt;li&gt;libcamera = Pi Camera on bullseye and newer&lt;/li&gt;
&lt;li&gt;V4L2 = USB webcams&lt;/li&gt;
&lt;/ul&gt;
&lt;/body&gt;
&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Driver</string>
         </property>
        </widget>
       </item>
       <item row="9" column="0">
        <widget class="QLabel" name="label_34">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Rotate the video one or more ways. This may not work with V4L2.&lt;/p&gt;&lt;p&gt;Often, only 0 or 180 degrees will work.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Rotate</string>
         </property>
        </widget>
       </item>
       <item row="14" column="0">
        <widget class="QLabel" name="label_32">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;H.264 profile to use with the encoder. The profile determines the overall quality of the stream at a given bitrate. Generally, better quality requires longer to encode and increases latency. It is recommended to use baseline (especially if using libx264 encoder) to minimize latency, however with hardware accelerated encoding (V4L2) or when using a Pi camera higher profiles may work.&lt;/p&gt;
&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Profile</string>
         </property>
        </widget>
       </item>
       <item row="22" column="2">
        <widget class="QLineEdit" name="txt_rtspkey">
         <property name="text">
          <string>stream</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QComboBox" name="combox_iomode">
         <item>
          <property name="text">
           <string>auto</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>rw</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>mmap</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>userptr</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dmabuf</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>dmabuf-import</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QComboBox" name="combox_driver">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <item>
          <property name="text">
           <string>raspicam</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>libcamera</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>v4l2</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QLineEdit" name="txt_width">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Width of the video mode. Must be a valid video mode along with the height and framerate settings.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>640</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_23">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Width for the video. Must be a supported video mode along with the setting for height and framerate.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Width</string>
         </property>
        </widget>
       </item>
       <item row="23" column="0">
        <spacer name="verticalSpacer_7">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="8" column="2">
        <widget class="QComboBox" name="combox_flip">
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Vertical</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Horizontal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Both</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="18" column="0">
        <widget class="QLabel" name="label_36">
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Network</string>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QLineEdit" name="txt_height">
         <property name="toolTip">
          <string>480</string>
         </property>
         <property name="text">
          <string>480</string>
         </property>
        </widget>
       </item>
       <item row="17" column="2">
        <widget class="QLineEdit" name="txt_quality">
         <property name="text">
          <string>50</string>
         </property>
        </widget>
       </item>
       <item row="21" column="0">
        <widget class="QLabel" name="label_39">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;TCP: The port to run the server on.&lt;/p&gt;&lt;p&gt;UDP: The port to send the frames to (at the given address).&lt;/p&gt;&lt;p&gt;RTSP: The port of the RTSP server&lt;/p&gt;&lt;p&gt;Default = 5008 for TCP or UDP; 8554 for RTSP&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Port</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_21">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;I/O mode to use with a V4L2 device. Generally, use auto. If latency is an issue, try dmabuf.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    IO Mode</string>
         </property>
        </widget>
       </item>
       <item row="20" column="0">
        <widget class="QLabel" name="label_38">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;For TCP, this is the address to run the server on. Use 0.0.0.0 to run the server on all the Pi's IP addresses (recommended).&lt;/p&gt;&lt;p&gt;For UDP, this is the address of the device to send frames to via UDP (the address of the computer you want to play the stream on).&lt;/p&gt;&lt;p&gt;For RTSP, this is the address of the running RTSP server (typically 127.0.0.1 as the server usually runs on the Pi).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Address</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="label_33">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flip the image from the camera one or more ways. This may not work with V4L2.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Flip</string>
         </property>
        </widget>
       </item>
       <item row="10" column="2">
        <widget class="QLineEdit" name="txt_gain">
         <property name="text">
          <string>10.0</string>
         </property>
        </widget>
       </item>
       <item row="16" column="0">
        <widget class="QLabel" name="label_29">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;For h264 streams, this is a desired bitrate of the video stream in bits / sec (ex 2048000 = 2kbits / sec). For mjpeg, this generally has no effect. However, when using the raspicam driver, this option has the same effect for mjpeg as it does for h264. In this case, the desired bitrate indirectly controls the jpeg compression quality. &lt;/p&gt;&lt;p&gt;Default = 2048000 = 2Mbits/sec&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>    Bitrate</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CamstreamDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CamstreamDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
